data/dynos_bin_read.cpp

```
@@ -117,10 +122,16 @@ char *DynOS_Read_Buffer(FILE* aFile, GfxData* aGfxData) {
 
 template <typename T>
 static void AppendNewNode(GfxData *aGfxData, DataNodes<T> &aNodes, const String &aName, String *&aDataName, Array<String> *&aDataTokens) {
-    DataNode<T> *_Node = New<DataNode<T>>();
-    _Node->mName = aName;
-    _Node->mModelIdentifier = aGfxData->mModelIdentifier;
-    aNodes.Add(_Node);
+    DataNode<T> *_Node = aNodes.FindExact(aName, aGfxData->mDataIdentifier);
+    if (_Node != NULL) {
+        PrintDataError("  ERROR: Node \"%s\" already exists for data identifier: %llX", aName.begin(), aGfxData->mDataIdentifier);
+    } else {
+        _Node = New<DataNode<T>>();
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)